src/EventListener/FormListener.php

```diff
@@ -0,0 +1,83 @@
+<?php
+namespace Bolt\EventListener;
+
+use Bolt\Config;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\HttpFoundation\Session\Storage\Handler\NativeFileSessionHandler;
+use Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage;
+
+/**
+ * Symfony Forms listener.
```

Comment:
OK, so the problem is this. If you're going to build a form using the DI:
```php
$this->app['form.factory']->createNamedBuilder($formname, $type, $data, $options)
    ->getForm();
```

That's all fine. The problem is that when the form is built, it accesses the current session, which by default sets the session cookie path to `/`, which applies it over the whole site…

So you need to pass a new storage provider to session, and that needs to have the `cookie_path` set to the relative URL you want the session cookie to apply to…

Now the fun part, Symfony Forms requires you to use `$builder->addEventSubscriber()` as unlike kernel events, they're not "on" by default.